Simplify spec of reordering and make it not flaky

alexander-cit opened the pull request:
## Checklist

- [x] Tests
<details>
<summary>More info</summary>

### Unit tests

Did you add relevant unit tests?

### E2E tests

Sometimes it can be more efficient to update E2E tests after CI has run them. If you know which ones to update, go ahead! E2E template cl2-back:

```bash
docker compose run --rm web bin/rails cl2_back:create_tenant[localhost,e2etests_template]
```

</details>

- [x] Prepared branch for code review
<details>
<summary>More info</summary>
Reviewed code to reduce unnecessary back and forth (removal of console.log, comments, ...)? Added comments to clarify code, emphasize what to pay attention to, etc.?
</details>

## How urgent is a code review?

Blocking the release.

alexander-cit committed:
Simplify spec of reordering and make it not flaky

alexander-cit commented:
I'm still investigating it. Just want to unblock the release.

sebastienhoorens reviewed (commented):
Indeed this test has been failing a lot recently. Do you have any idea what's going wrong? I would prefer if we can keep guaranteeing that publications get reordered as expected. Perhaps there's something random in the created publications before this test is executed? Could we instead try something like deleting all publications and adding new ones in this spec to see if that works?

Something like this could be helpful to test if the spec is no longer flaky:
`for i in `seq 30`; do docker-compose run --rm web rspec ./spec/acceptance/admin_publications_spec.rb; [[ ! $? = 0 ]] && break ; done`

If not, how would we follow up on fixing this test if we would proceed with commenting this out? Because I'm afraid if we would just log a bug in the backlog, it will never get taken up.

alexander-cit commented:
Thanks for the command, it's really useful. But unfortunately, locally it passed successfully. And it looks like on the CI it failed almost every time.

But you're absolutely right, we have to fix it. First, I was too lazy to do it but then I realized we really need it. Thanks for the motivation! :slightly_smiling_face: 
I'll try to simplify the test.

sebastienhoorens commented:
Strange that it passes locally... In that case, perhaps there are other specs that impact the initial state of the spec? This has happened in the past, and that's when you can have a situation where specs consistently succeed when running just the spec file, but fail when running all specs together (depending on the order in which they were ran).

alexander-cit commented:
@sebastienhoorens I rewrote and simplified the spec. 
It was failing because there were several publications with the same ordering. And sometimes the test used the wrong publication.

sebastienhoorens reviewed (approved):
Great you found it!

adessy reviewed (approved):
Would it make sense to add a uniqueness constraint at the DB level to make sure there is no duplicated positions?

alexander-cit commented:
> It was failing because there were several publications with the same ordering

@adessy these publications belonged to different parents (`acts_as_list` uses `parent_id` scope), but the spec didn't respect `parent_id`. So, the DB was still consistent, it's just the spec that was wrong.
